src/synopsys-action/synopsys-bridge.ts

```diff
@@ -126,53 +126,73 @@ export class SynopsysBridge {
       await extractZipped(downloadResponse.filePath, extractZippedFilePath)
       info('Download and configuration of Synopsys Bridge completed')
     } catch (e) {
-      const error = (e as Error).message
+      const errorObject = (e as Error).message
       await cleanupTempDir(tempDir)
-      if (error.includes('404') || error.toLowerCase().includes('invalid url')) {
+      if (errorObject.includes('404') || errorObject.toLowerCase().includes('invalid url')) {
         let os = ''
         if (process.env['RUNNER_OS']) {
           os = process.env['RUNNER_OS']
         }
         return Promise.reject(new Error('Provided Bridge url is not valid for the configured '.concat(os, ' runner')))
-      } else if (error.toLowerCase().includes('empty')) {
+      } else if (errorObject.toLowerCase().includes('empty')) {
         return Promise.reject(new Error('Provided Bridge URL cannot be empty'))
       } else {
-        return Promise.reject(new Error(error))
+        return Promise.reject(new Error(errorObject))
       }
     }
   }
 
   async prepareCommand(tempDir: string): Promise<string> {
     try {
       let formattedCommand = ''
+
+      const paramsMap = new Map()
+      paramsMap.set(constants.POLARIS_SERVER_URL_KEY, inputs.POLARIS_SERVER_URL)
+      paramsMap.set(constants.COVERITY_URL_KEY, inputs.COVERITY_URL)
+      paramsMap.set(constants.BLACKDUCK_URL_KEY, inputs.BLACKDUCK_URL)
+      const invalidParams: string[] = isNullOrEmpty(paramsMap)
+      if (invalidParams.length === 3) {
+        return Promise.reject(new Error('Requires at least one scan type: ('.concat(constants.POLARIS_SERVER_URL_KEY).concat(',').concat(constants.COVERITY_URL_KEY).concat(',').concat(constants.BLACKDUCK_URL_KEY).concat(')')))
+      }
       // validating and preparing command for polaris
-      if (validatePolarisInputs()) {
+      const polarisErrors: string[] = validatePolarisInputs()
+      if (polarisErrors.length === 0 && inputs.POLARIS_SERVER_URL) {
         const polarisCommandFormatter = new SynopsysToolsParameter(tempDir)
         formattedCommand = formattedCommand.concat(polarisCommandFormatter.getFormattedCommandForPolaris())
       }
 
       // validating and preparing command for coverity
-      if (validateCoverityInputs()) {
+      const coverityErrors: string[] = validateCoverityInputs()
+      if (coverityErrors.length === 0 && inputs.COVERITY_PASSPHRASE) {
```

Comment:
Why are we checking for coverity_passphrase here instead coverity url?

Reply:
changed to coverity_url